processor/tailsamplingprocessor/config.go

```diff
@@ -66,8 +66,15 @@ type NumericAttributeCfg struct {
 type StringAttributeCfg struct {
 	// Tag that the filter is going to be matching against.
 	Key string `mapstructure:"key"`
-	// Values is the set of attribute values that if any is equal to the actual attribute value to be considered a match.
+	// Values indicate the set of values or regular expressions to use when matching against attribute values.
+	// StringAttribute Policy will apply exact value match on Values unless RegexMatchEnabled is true.
 	Values []string `mapstructure:"values"`
+	// RegexMatchEnabled determines whether match attribute values by regexp string.
+	RegexMatchEnabled bool `mapstructure:"enabled_regex_matching"`
```

Comment:
Maybe we can nest the regex confing into its own struct like 

```go
type StringAttributeCfg struct {
     Regex StringAttributeRegexCfg // If a valid regex config is provided, Values is ignored and Regex.Expressions is used for matching.
}

type StringAttributeRegxCfg struct {
      Expressions []string
      CacheSize int
}
```

btw: Do we want to have a hybrid mode, i.e. first match fixed value, then do regexp match, in that case there can be a mode config with enums like `ValueOnly` `ValueAndRegex`, `RegexOnly` etc.

Reply:
Note that this code is in the hot path: every attribute of every span of every trace will reach this code. It has to be rock solid and do only what is needed. I don't think it should do regexp match at all if not explicitly requested by the user. In fact, I think a separate policy is still a better idea.